tools/universe/package_publisher.py

```diff
@@ -0,0 +1,163 @@
+#!/usr/bin/env python3
+
+import base64
+import difflib
+import http.client
+import json
+import logging
+import os
+import shutil
+
+log = logging.getLogger(__name__)
+logging.basicConfig(level=logging.DEBUG, format="%(message)s")
+
+class UniversePackagePublisher(object):
+    '''Creates a PR for a release against the Universe repository at http://github.com/mesosphere/universe.
+    '''
+
+    def __init__(self, package_name, package_version, commit_desc, dry_run=False):
+        self._pkg_name = package_name
+        self._pkg_version = package_version
+        self._pr_title = 'Release {} {} (automated commit)\n\n'.format(self._pkg_name, self._pkg_version)
+        self._commit_desc = commit_desc
+        self._dry_run = dry_run
+
+        # Optional configuration via envvars:
+        self._release_branch = os.environ.get('RELEASE_BRANCH', 'version-3.x')
+        self._github_user = os.environ.get('GITHUB_USER', 'mesosphere-ci')
+        self._github_token = os.environ.get('GITHUB_TOKEN', None)
+        if self._github_token is None:
+            raise Exception('GITHUB_TOKEN is required: Credential to create a PR against Universe')
+        self._enc_github_token = base64.encodestring(self._github_token.encode('utf-8')).decode('utf-8').rstrip('\n')
+        self._release_universe_repo = os.environ.get('RELEASE_UNIVERSE_REPO', 'mesosphere/universe')
+
+
+    def _create_universe_branch(self, scratchdir, pkgdir):
+        branch = 'automated/release_{}_{}_{}'.format(
+            self._pkg_name,
+            self._pkg_version,
+            base64.b64encode(os.urandom(4)).decode('utf-8').rstrip('='))
+
+        # check out the repo, create a new local branch:
+        ret = os.system(' && '.join([
+            'cd {}'.format(scratchdir),
+            'git clone --depth 1 --branch {} https://{}:{}@github.com/{} universe'.format(
+                self._release_branch, self._github_user, self._github_token, self._release_universe_repo),
+            'cd universe',
+            'git config --local user.email jenkins@mesosphere.com',
+            'git config --local user.name release_builder.py',
+            'git checkout -b {}'.format(branch)]))
+        if ret != 0:
+            raise Exception('Failed to create local Universe git branch {}.'.format(branch))
+        universe_repo = os.path.join(scratchdir, 'universe')
+        repo_pkg_base = os.path.join(
+            universe_repo,
+            'repo',
+            'packages',
+            self._pkg_name[0].upper(),
+            self._pkg_name)
+
+        # find the prior release number:
+        lastnum = -1
+        if not os.path.exists(repo_pkg_base):
```

Comment:
There is a package manager call that gives the latest release number. Is there a reason that we don't use that here? Or is it because we anyway have to create a PR?